android/src/main/java/com/tailscale/ipn/DnsConfig.java

```diff
@@ -102,260 +42,22 @@ public DnsConfig(Context ctx) {
 	//
 	// an empty string means the current DNS configuration could not be retrieved.
 	String getDnsConfigAsString() {
-		String s = getDnsConfigFromLinkProperties();
-		if (!s.trim().isEmpty()) {
-			return s;
-		}
-		if (android.os.Build.VERSION.SDK_INT >= 23) {
-			// If ConnectivityManager.getAllNetworks() works, it is the
-			// authoritative mechanism and we rely on it. The other methods
-			// which follow involve more compromises.
-			return "";
-		}
-
-		s = getDnsServersFromSystemProperties();
+		String s = getDnsConfigs();
 		if (!s.trim().isEmpty()) {
```

Comment:
unnecessary? We're now returning empty string regardless

Reply:
We return the string if it's nonempty

Reply:
Sure, but if the string is empty, we return `""` below. So, you could skip the conditional and always return `getDnsConfigs().trim()` for the same effect